includes/Core/Permissions/Permissions.php

```diff
@@ -428,6 +483,23 @@ private function check_view_shared_dashboard_capability( $user_id ) {
 		return array();
 	}
 
+	/**
+	 * Checks if the VIEW_AUTHENTICATED_DASHBOARD capability is allowed for the user.
+	 *
+	 * Allows access to the VIEW_AUTHENTICATED_DASHBOARD capability if the user is authenticated.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $user_id User ID of the user the capability is checked for.
+	 * @return array Array with a 'do_not_allow' element if checks fail, otherise returns AUTHENTICATE capability.
+	 */
+	private function check_view_authenticated_dashboard_capability( $user_id ) {
+		if ( $this->is_user_authenticated( $user_id ) && $this->authentication->is_setup_completed() ) {
```

Comment:
Good catch on adding the check for setup completed here. 👍 

The other cases are covered by that generic condition in our `map_meta_cap` callback, as almost all capabilities in `$base_to_core` or `$meta_to_core` will not be granted unless the setup is complete.